consensus/hotstuff/cruisectl/block_time_controller.go

```diff
@@ -328,32 +332,28 @@ func (ctl *BlockTimeController) processIncorporatedBlock(tb TimedBlock) error {
 // No errors are expected during normal operation.
 func (ctl *BlockTimeController) checkForEpochTransition(tb TimedBlock) error {
 	view := tb.Block.View
-	if view <= ctl.curEpochFinalView { // prevalent case: we are still within the current epoch
+	if view <= ctl.currentEpochTiming.finalView { // prevalent case: we are still within the current epoch
 		return nil
 	}
 
 	// sanity checks, since we are beyond the final view of the most recently processed epoch:
-	if ctl.nextEpochFinalView == nil { // final view of epoch we are entering should be known
-		return fmt.Errorf("cannot transition without nextEpochFinalView set")
-	}
-	if ctl.nextEpochTargetEndTime == nil {
-		return fmt.Errorf("cannot transition without nextEpochTargetEndTime set")
+	if ctl.nextEpochTiming == nil { // next epoch timing not initialized
+		return fmt.Errorf("sanity check failed: cannot transition without next epoch timing initialized")
 	}
-	if ctl.nextEpochTargetDuration == nil {
-		return fmt.Errorf("cannot transition without nextEpochTargetDuration set")
+	if !ctl.currentEpochTiming.isFollowedBy(ctl.nextEpochTiming) { // non consecutive epochs
+		return fmt.Errorf("sanity check failed: invalid epoch transition current epoch (final view: %d) is not followed by next epoch (first view: %d)",
+			ctl.currentEpochTiming.finalView, ctl.nextEpochTiming.firstView)
 	}
-	if view > *ctl.nextEpochFinalView { // the block's view should be within the upcoming epoch
+	if view > ctl.nextEpochTiming.finalView { // the block's view should be within the upcoming epoch
 		return fmt.Errorf("sanity check failed: curView %d is beyond both current epoch (final view %d) and next epoch (final view %d)",
-			view, ctl.curEpochFinalView, *ctl.nextEpochFinalView)
+			view, ctl.currentEpochTiming.finalView, ctl.nextEpochTiming.finalView)
 	}
 
-	ctl.curEpochFirstView = ctl.curEpochFinalView + 1
-	ctl.curEpochFinalView = *ctl.nextEpochFinalView
-	ctl.curEpochTargetDuration = *ctl.nextEpochTargetDuration
-	ctl.curEpochTargetEndTime = *ctl.nextEpochTargetEndTime
-	ctl.nextEpochFinalView = nil
-	ctl.nextEpochTargetDuration = nil
-	ctl.nextEpochTargetEndTime = nil
+	ctl.currentEpochTiming.firstView = ctl.nextEpochTiming.firstView
+	ctl.currentEpochTiming.finalView = ctl.nextEpochTiming.finalView
+	ctl.currentEpochTiming.targetDuration = ctl.nextEpochTiming.targetDuration
+	ctl.currentEpochTiming.targetEndTime = ctl.nextEpochTiming.targetEndTime
```

Comment:
```suggestion
	ctl.currentEpochTiming = *ctl.nextEpochTiming
```
Optionally, a more concise way to move the data over.